copy/single.go

```diff
@@ -131,10 +137,13 @@ func (c *copier) copySingleImage(ctx context.Context, unparsedImage *image.Unpar
 		src:             src,
 		// diffIDsAreNeeded is computed later
 		cannotModifyManifestReason:    cannotModifyManifestReason,
-		requireCompressionFormatMatch: requireCompressionFormatMatch,
+		requireCompressionFormatMatch: opts.requireCompressionFormatMatch,
 	}
-	if c.options.DestinationCtx != nil {
+	if opts.compressionFormat != nil {
 		// Note that compressionFormat and compressionLevel can be nil.
```

Comment:
This comment belongs to the old `DestinationCtx` case. In the new one, we have ensured `opts.compressionFormat != nil` just above.